MozziGuts_impl_ARDUINOFSP.hpp

```diff
@@ -0,0 +1,227 @@
+ */
+
+#if !(IS_FSP())
+#  error "Wrong implementation included for this platform"
+#endif
+
+#include <FspTimer.h>
+
+
+////// BEGIN analog input code ////////
+
+//#define MOZZI_FAST_ANALOG_IMPLEMENTED
+
+#define getADCReading() 0
+
+
+#define channelNumToIndex(channel) channel
+uint8_t adcPinToChannelNum(uint8_t pin) {
+  return pin;
+}
+
+/** NOTE: Code needed to trigger a conversion on a new channel */
+void adcStartConversion(uint8_t channel) {
+#warning Fast analog read not implemented on this platform
+}
+
+/** NOTE: Code needed to trigger a subsequent conversion on the latest channel. If your platform has no special code for it, you should store the channel from
+ *  adcStartConversion(), and simply call adcStartConversion(previous_channel), here. */
+void startSecondADCReadOnCurrentChannel() {
+#warning Fast analog read not implemented on this platform
+}
+
+/** NOTE: Code needed to set up faster than usual analog reads, e.g. specifying the number of CPU cycles that the ADC waits for the result to stabilize.
+ *  This particular function is not super important, so may be ok to leave empty, at least, if the ADC is fast enough by default. */
+void setupFastAnalogRead(int8_t speed) {
+#warning Fast analog read not implemented on this platform
+}
+
+/** NOTE: Code needed to initialize the ADC for asynchronous reads. Typically involves setting up an interrupt handler for when conversion is done, and
+ *  possibly calibration. */
+void setupMozziADC(int8_t speed) {
+#warning Fast analog read not implemented on this platform
+}
+
+/* NOTE: Most platforms call a specific function/ISR when conversion is complete. Provide this function, here.
+ * From inside its body, simply call advanceADCStep(). E.g.:
+ void stm32_adc_eoc_handler() {
+ advanceADCStep();
+ }
+*/
+////// END analog input code ////////
+
+
+
+//// BEGIN AUDIO OUTPUT code ///////
+
+
+FspTimer timer;
+
+#if (EXTERNAL_AUDIO_OUTPUT != true) // otherwise, the last stage - audioOutput() - will be provided by the user
+FspTimer timer_dac;
+#include <dac.h>
+#include <FspTimer.h>
+#include <r_dtc.h>
+
+
+/*
+Good parts of the following are from analogWave example class
+*/
+
+volatile uint32_t pin;
+CircularBuffer<uint16_t> output_buffer;
+uint8_t dac_bits;
+dtc_instance_ctrl_t dtc_ctrl;
+transfer_info_t dtc_info;
+dtc_extended_cfg_t dtc_cfg_extend;
+transfer_cfg_t dtc_cfg;
+
+////////////// DAC CREATION AND FIRST INIT ///////////////////
+void dac_creation(pin_size_t pinN) {
+  if (IS_DAC(pinN)) {
+    auto cfg_dac = getPinCfgs(pinN, PIN_CFG_REQ_DAC);
+    pin = cfg_dac[0];
+    if (IS_DAC_8BIT(pin)) {
+      dac_bits = 8;
+      dtc_info.transfer_settings_word_b.size = TRANSFER_SIZE_1_BYTE;
+      if (GET_CHANNEL(pin) == 0) {
+#ifdef DAC_ADDRESS_8_CH0
+        dtc_info.p_dest = (void *)DAC_ADDRESS_8_CH0;
+#endif
+      } else if (GET_CHANNEL(pin) == 1) {
+#ifdef DAC_ADDRESS_8_CH1
+        dtc_info.p_dest = (void *)DAC_ADDRESS_8_CH1;
+#endif
+      }
+    } else {
+      dac_bits = 12;
```

Comment:
Actually, this is drawn from the DAC capacity. I think it would be smarted to state that this port is 12bits, and 12 bits only (this function gets it from the pin it uses to output Audio, not sure how configurable it is).